src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -21960,14 +21960,275 @@ public void Main(string s)
+                //             s.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(9, 13),
+                // (17,13): warning CS8602: Dereference of a possibly null reference.
+                //             s.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(17, 13)
+                );
+        }
+
+        [Fact, WorkItem(29855, "https://github.com/dotnet/roslyn/issues/29855")]
+        public void IsNullNotEqualsTrue()
+        {
+            var c = CreateNullableCompilation(@"
+class C
+{
+    void M(string s)
+    {
+        if ((s is null) != true)
+            s.ToString();
+        else
+            s.ToString(); // 1
+    }
+
+    void M2(string s)
+    {
+        if (true != (s is null))
+            s.ToString();
+        else
+            s.ToString(); // 2
+    }
+}
+");
+
+            c.VerifyDiagnostics(
+                // (9,13): warning CS8602: Dereference of a possibly null reference.
+                //             s.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(9, 13),
+                // (17,13): warning CS8602: Dereference of a possibly null reference.
+                //             s.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(17, 13)
+                );
+        }
+
+        [Fact, WorkItem(29855, "https://github.com/dotnet/roslyn/issues/29855")]
+        public void IsNullNotEqualsFalse()
+        {
+            var c = CreateNullableCompilation(@"
+class C
+{
+    void M(string s)
+    {
+        if ((s is null) != false)
+            s.ToString(); // 1
+        else
+            s.ToString();
+    }
+
+    void M2(string s)
+    {
+        if (false != (s is null))
+            s.ToString(); // 2
+        else
+            s.ToString();
+    }
+}
+");
+
+            c.VerifyDiagnostics(
+                // (7,13): warning CS8602: Dereference of a possibly null reference.
+                //             s.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(7, 13),
+                // (15,13): warning CS8602: Dereference of a possibly null reference.
+                //             s.ToString(); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(15, 13)
+                );
+        }
+
+        [Fact, WorkItem(29855, "https://github.com/dotnet/roslyn/issues/29855")]
+        public void OnlySwapConditionalStatesWhenInConditionalState()
+        {
+            var c = CreateNullableCompilation(@"
+class C
+{
+    void M(bool b)
+    {
+        if (b == (true != b))
+        {
+        }
+    }
+}
+");
+
+            c.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(41107, "https://github.com/dotnet/roslyn/issues/41107")]
+        public void NullConditionalEqualsTrue()
```

Comment:
>EqualsTrue [](start = 35, length = 10)

`NotNull`?

Reply:
I didn't understand. We're comparing a null-conditional to true, thus the test name.